jaggery-modules/rxt/module/scripts/core/core.js

```diff
@@ -622,16 +619,41 @@ var core = {};
             log.error('Unable to locate the rxt definition for type: ' + type + ' in order to return taxonomy ');
             throw 'Unable to locate the rxt definition for type: ' + type + ' in order to return taxonomy ';
         }
+        return taxonomyAvailability(type, rxtDefinition);
+
+    };
+
+    /**
+     * This method will return the available taxonomies for a given asset type by reading tenant specific map
+     * @param type asset type
+     * @param rxtDefinition specific for a tenant
+     * @returns  array of available taxonomies
+     */
+    var taxonomyAvailability = function (type, rxtDefinition) {
+        var carbon = require('carbon');
+        var TaxonomyService = carbon.server.osgiService('org.wso2.carbon.governance.taxonomy.services.' +
+            'ITaxonomyServices');
+        var taxonomies;
+        var taxonomyArray = [];
         if (rxtDefinition) {
             if (rxtDefinition.taxonomies && (rxtDefinition.taxonomies[0]) && (rxtDefinition.taxonomies[0].taxonomies)) {
-                return rxtDefinition.taxonomies[0].taxonomies || '';
+                taxonomies = rxtDefinition.taxonomies[0].taxonomies || '';
             }
         }
-        if (log.isDebugEnabled()) {
-            log.debug('Unable to locate a taxonomy property in RXT in order retrieve default taxonomy name for ' + type);
+
+        if (taxonomies && (taxonomies = taxonomies.split(","))) {
+
+            for (var i = 0; i < taxonomies.length; i++) {
+                if (TaxonomyService.getTaxonomy(taxonomies[i])) {
+                    taxonomyArray.push(taxonomies[i]);
+                } else {
+                    log.error("The taxonomy name : " + taxonomies[i] + " not matched with rxt definition. " +
```

Comment:
Might not need to keep a space before the colon. Shouldn't this sentence be written in present tense?